src/java/com/puppetlabs/http/client/Sync.java

```diff
@@ -31,7 +31,12 @@ private static RequestOptions extractRequestOptions(SimpleRequestOptions simpleO
         Object body = simpleOptions.getBody();
         boolean decompressBody = simpleOptions.getDecompressBody();
         ResponseBodyType as = simpleOptions.getAs();
-        return new RequestOptions(uri, headers, body, decompressBody, as);
+        CompressType requestBodyDecompression =
+                simpleOptions.getCompressRequestBody();
+        RequestOptions requestOptions = new RequestOptions(
+                uri, headers, body, decompressBody, as);
+        requestOptions.setCompressRequestBody(requestBodyDecompression);
```

Comment:
`RequestOptions` is public API at this point so I don't think I could safely just change the existing constructor to take an additional argument without potentially breaking an existing client.  I could add an overloaded constructor which takes the extra argument.  Since I'm not sure how often this option would be used by consumers outside of the one I'm thinking of today, the PuppetDB terminus, just having a way to access it via a setter without creating another constructor seemed like a better way to go for now.  Does that make sense?